airbyte-config/config-models/src/main/java/io/airbyte/config/Configs.java

```diff
@@ -541,13 +559,77 @@ public interface Configs {
    */
   boolean shouldRunConnectionManagerWorkflows();
 
+  /**
+   * Define if the worker is operating as a Control Plane worker, a Data Plane worker, or as a
+   * combined worker. - Control plane workers process tasks related to control-flow, like scheduling
+   * and routing. - Data plane workers process tasks related to data syncing, and are typically
+   * deployed outside of Airbyte's internal network. - Combined workers process all tasks, and are the
+   * default for OSS deployments. - Internal-use only.
+   */
+  WorkerPlane getWorkerPlane();
+
+  // Worker - Control Plane configs
+
+  /**
+   * Return true if the WorkerPlane is CONTROL_PLANE or COMBINED. Internal-use only.
+   */
+  boolean isControlPlaneWorker();
```

Comment:
is the result of this call dependent on the `getWorkerPlane` call?

Reply:
The `EnvConfigs` implementation of `isControlPlaneWorker` and `isDataPlaneWorker` both call `getWorkerPlane()` under the hood. Perhaps I should note the relationship between these in the javadoc here